dm/tests/shardddl4_1/run.sh

```diff
@@ -970,49 +970,78 @@ function DM_155_CASE {
 	sleep 1
 	random_restart 3
 
-	run_sql_source1 "insert into ${shardddl1}.${tb1} values(25,25,25,25,25);"
-	run_sql_source2 "insert into ${shardddl1}.${tb1} values(26,26,26,26,26);"
-	run_sql_source2 "insert into ${shardddl1}.${tb2} values(27,27,27);"
-
+	# make sure source1 has received conflict ddl and handled
+	source1worker=$($PWD/bin/dmctl.test DEVEL --master-addr "127.0.0.1:$MASTER_PORT" query-status test -s mysql-replica-01 |
+		grep 'worker' | awk -F: '{print $2}' | cut -d'"' -f 2)
+	check_log_contain_with_retry "skip conflict ddls in optimistic shard mode.*ALTER TABLE \`${shardddl1}\`.\`${tb1}\` CHANGE COLUMN \`c\` \`b\` INT" \
+		$WORK_DIR/$source1worker/log/dm-worker.log
 	run_sql_source2 "alter table ${shardddl1}.${tb2} change c b int;"
-	run_sql_source1 "insert into ${shardddl1}.${tb1} values(28,28,28,28,28);"
-	run_sql_source2 "insert into ${shardddl1}.${tb1} values(29,29,29,29,29);" # source2 won't redirect until it receives new event
-	run_sql_source2 "insert into ${shardddl1}.${tb2} values(30,30,30);"
 
+	# make sure source2 has handled tb2 ddl and trigger redirection
+	check_log_contain_with_retry "finish to handle ddls in optimistic shard mode.*alter table ${shardddl1}.${tb2} change c b int" \
```

Comment:
if we use variable source1worker, here we should also use source2worker.

I remember then cluster initialized, source1 should be bound to worker1, maybe we don't need these dynamic variable

Reply:
addressed in https://github.com/pingcap/tiflow/pull/5868/commits/f05b84de1c554ec2edf17da4e7c3b8e20b0d963a